src/Integration.Vsix/SonarAnalyzerManager.cs

```diff
@@ -61,35 +62,55 @@ public sealed class SonarAnalyzerManager
             this.activeSolutionBoundTracker = activeSolutionBoundTracker;
             this.workspace = workspace;
 
-            SonarAnalysisContext.ShouldAnalysisBeDisabled = tree => ShouldAnalysisBeDisabledOnTree(tree);
+            SonarAnalysisContext.ShouldExecuteRuleFunc = ShouldExecuteRule;
         }
 
-        private bool ShouldAnalysisBeDisabledOnTree(SyntaxTree tree)
+        private bool ShouldExecuteRule(AnalysisRunContext context)
         {
-            if (tree == null)
+            if (context.SyntaxTree == null)
             {
-                return false;
+                return true;
             }
 
-            IEnumerable<AnalyzerReference> references = workspace?.CurrentSolution?.GetDocument(tree)?.Project?.AnalyzerReferences;
-            ProjectAnalyzerStatus projectAnalyzerStatus = GetProjectAnalyzerConflictStatus(references);
+            var references = this.workspace.CurrentSolution?.GetDocument(context.SyntaxTree)?.Project?.AnalyzerReferences;
+            var projectAnalyzerStatus = GetProjectAnalyzerConflictStatus(references);
 
-            return HasConflictingAnalyzerReference(projectAnalyzerStatus) ||
-                this.GetIsBoundWithoutAnalyzer(projectAnalyzerStatus);
+            return !HasConflictingAnalyzerReference(projectAnalyzerStatus) &&
+                !GetIsBoundWithoutAnalyzer(projectAnalyzerStatus) &&
+                HasAnyRuleEnabled(context.SupportedDiagnostics);
         }
 
-        internal /*for testing purposes*/ bool GetIsBoundWithoutAnalyzer(ProjectAnalyzerStatus projectAnalyzerStatus)
+        internal /*for testing purposes*/ bool HasAnyRuleEnabled(IEnumerable<DiagnosticDescriptor> supportedDiagnostics)
         {
-            return projectAnalyzerStatus == ProjectAnalyzerStatus.NoAnalyzer &&
+            Debug.Assert(supportedDiagnostics != null, "Not expecting a null list of diagnostics");
+
+            switch (this.activeSolutionBoundTracker.CurrentMode)
+            {
+                case NewConnectedMode.SonarLintMode.Standalone:
+                    // For now the standalone is not configurable so we only enable rules part of SonarWay profile
+                    return supportedDiagnostics.Any(d => d.CustomTags.Contains(DiagnosticTagsHelper.SonarWayTag));
+
+                case NewConnectedMode.SonarLintMode.LegacyConnected:
+                    // Ruleset is used to decide whether or not the rule should be enabled
+                    return true;
+
+                case NewConnectedMode.SonarLintMode.Connected:
+                    // TODO: Call the class responsible of the new rule set to decide whether any rule is enabled
+                    return true;
+
+                default:
+                    Debug.Fail("Unhandled SonarLintMode");
```

Comment:
I could either:
1/ Not handle the case.
2/ throw (but I think the extension should avoid failing)
3/ [current solution] fail in debug and avoid the analysis to be run.

WDYT?

Reply:
Write a test and throw :)

Reply:
But throw will kill all features from the extension. Is that ok?

Reply:
It won't kill all the features in the extension, but it will probably cause VS shows the _Analyzer X has thrown an exception_ message for every single analyzer (unless it's smart enough not to try to display them all!)
I'd assert and fail-safe, so either disable the rules, or treat as standalone.